### PR TITLE
366 generate python api documentation

### PR DESCRIPTION
**Pull request by quaat**
Generates python-documentation with Sphinx from CMake

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -299,6 +299,7 @@ The sources can be cloned from GitHub
   - [cppcheck][13], optional, used for static code analysis.
   - [mongomock][mongomock], optional, used for testing mongodb storage plugin.
   - librdf development libraries, optional, needed by librdf storage plugin.
+
```
Comment (CasperWA) [inline README.md:302]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -379,6 +380,17 @@ system defined in Dockerfile.
...
 ### Build documentation
+
+#### Build Python Documentation
+
+DLite uses sphinx to generate documentation from Python source
+code. Ensure the correct virtual environment is set up and install the
+requirements `pip install -r requirements_doc.txt`
+
+
+
```
Comment (CasperWA) [inline README.md:391]:
```suggestion
```

File: pydoc/index.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
+DLite Python API
```
Comment (CasperWA) [inline pydoc/index.rst:1]: This should be updated to be a TOC that is divides into 1) C API 2) Python API 3) Fortran API.
I think at least that there's a Fortran API as well?

This point can be made into a separate issue and updated later.

**Review by CasperWA — APPROVED** [review]
Looks good.
Please create the issue as commented below and the suggested changes are not necessary but would clean up the README slightly even so.

File: pydoc/index.rst (hunk 3 above)
Comment (quaat) [reply pydoc/index.rst:1]: Follow-up task created: #368 Extend Sphinx-generated documentation to include C/C++ and Fortran API

**Review by CasperWA — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -379,6 +379,15 @@ system defined in Dockerfile.
...
 
 ### Build documentation
+
+#### Build Python Documentation
+
+DLite uses sphinx to generate documentation from Python source
+code. Ensure the correct virtual environment is set up and install the
+requirements `pip install -r requirements_doc.txt`
+
+#### Build C++ Documentation
```
Comment (jesper-friis) [inline README.md:389]:
```suggestion
#### Build C API Documentation
```

**Comment by jesper-friis** [comment]
I guess that we can remove `pydoc/Doxyfile.in` since we are not using doxygen for the python documentation but sphinx?

File: pydoc/conf.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# For the full list of built-in configuration values, see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = 'Knowledge Base'
```
Comment (jesper-friis) [inline pydoc/conf.py:9]: DLite

**Comment by jesper-friis** [comment]
> I guess that we can remove `pydoc/Doxyfile.in` since we are not using doxygen for the python documentation but sphinx?

Maybe I am wrong. There are some comments in [pydoc/conf.py](https://github.com/SINTEF/dlite/pull/367/files#diff-8b902fa12775457ab95db99305160943e5485afc011327f5853c5e45eb8e2ae0) that seems to indicate that you are using doxygen via sphinx...

**Review by jesper-friis — APPROVED** [review]
Looks good. I added a few comments that you can check if they make sense.

**Comment by quaat** [comment]
> I guess that we can remove `pydoc/Doxyfile.in` since we are not using doxygen for the python documentation but sphinx?

Well, the intention was to run Sphinx with Doxygen and generate all documentation together, but a follow-up task has been created for this issue.

**Comment by quaat** [comment]
>

**Pull request merged**